chore(android)(9_0_X): auto-download NDK r21c for module builds if needed

jquick-axway opened the pull request:
**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-27939
- https://jira.appcelerator.org/browse/TIMOB-27776

**Cherry-pick of PR:** #11755

jquick-axway committed:
chore(android): auto-download NDK r21c for module builds

- This NDK version was notarized for macOS Catalina. Prevents security popups when doing module builds.

build commented:
<!--
  1 failure:  Tests have failed...
  0 warning: 
  2 messages
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:x: 1 tests have failed
There are 1 tests failing and 225 skipped out of 2477 total tests.</td>
    </tr>
  </tbody>
</table>

### Tests:

<table><tr><th>Classname</th><th>Name</th><th>Time</th><th>Error</th></tr>
<tr><td>ios.ipad.Titanium</td><td>buildDate (13.4.1)</td><td>0.001</td><td><pre>Error: expected '6/5/2020 00:39' to match /[01]?\d\/[0123]\d\/20\d{2} \d{2}:\d{2}/
fail@file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/node_modules/should/cjs/should.js:275:19
value@file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/node_modules/should/cjs/should.js:356:13
file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/ti.test.js:32:27
file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/ti-mocha.js:6535:54167
file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/ti-mocha.js:6535:58607
file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/ti-mocha.js:6535:59452
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/ti-mocha.js:6535:58015
file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/ti-mocha.js:6535:57979
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/ti-mocha.js:6535:57667
file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/ti-mocha.js:6535:57763
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/455E2DDA-2DF5-4A79-975B-B71735C2C9E6/data/Containers/Bundle/Application/181F4E76-EC9B-4419-B13B-769D73286F07/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
</table>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ea2ca087bd777e090afda01e4971b7b71c7e93d5
</p>

jquick-axway committed:
fix(android): bad file descriptor errors with NDK v21 builds

- Compiling with Android NDK 21 and higher would cause several harmless "Bad file descriptor" errors to be logged.
- Caused by NDK 21 setting "--output-sync" argument default. Setting it to "none" restores old behavior.

jquick-axway committed:
fix(android): only use --output-sync for NDK 21+

- Using NDK command line argument --output-sync will cause a build failure on NDK 19 and lower. And we don't need to pass it into NDK 20.

sgtcoolguy committed:
Merge branch '9_0_X' into TIMOB-27939-9_0_X

sgtcoolguy committed:
Merge branch '9_0_X' into TIMOB-27939-9_0_X

jquick-axway committed:
chore(android): removed TIMOB-27939 commented-out code

garymathews committed:
Merge branch '9_0_X' into TIMOB-27939-9_0_X

garymathews reviewed (approved):
CR: **PASS**

ssjsamir commented:
**FR Passed:** No longer see `Bad file descriptor` errors and able to install NDK if no other NDKs are found. 

**Test Environment**
```
MacOS Catalina: 10.15.5 Beta
Xcode: 11.4
Java Version: 1.8.0_131
Android NDK: 21.1.6273396-beta2
Node.js: 10.16.3
""NPM":"5.0.0","CLI":"8.0.0""
Andoird Pixl xl 7.1.1 Emulator 
```
